sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,37 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, is_directed_acyclic_graph
```

Comment:
Networkx is a very mature library, and I think you used it well.
Do you know how Airflow identifies cyclic dependencies at the moment? I'd be curious to see how both approaches relate.

Reply:
I don't know, but I can try to find out. @kaxil do you know?

Reply:
In 2.4.0, [this](https://github.com/apache/airflow/blob/2.4.0/airflow/utils/dag_cycle_tester.py#L49-L83) is how it detects cycles. So it does not use any external package for that.